### PR TITLE
[mypy] [9877] Reduce mypy errors

### PR DESCRIPTION
**Pull request by rodrigc**
https://twistedmatrix.com/trac/ticket/9877

### CODE REVIEW

File: src/twisted/internet/base.py [hunk 1]
```diff
@@ -719,7 +719,9 @@ def fireSystemEvent(self, eventType):
             event.fireEvent()
 
 
-    def addSystemEventTrigger(self, _phase, _eventType, _f, *args, **kw):
+    # We need to ignore mypy errors here because the '_f' parameter is
+    # not in IReactorCore.addSystemEventTrigger
+    def addSystemEventTrigger(self, _phase, _eventType, _f, *args, **kw):  # type: ignore[override]  # noqa
```
Comment (adiroiban) [inline src/twisted/internet/base.py]: I think that this should follow the signature from the interface: `def addSystemEventTrigger(phase, eventType, callable, *args, **kw):` and rename `_phase` -> `phase` and `_f` -> `callable`

File: src/twisted/internet/base.py [hunk 2]
```diff
@@ -785,7 +787,9 @@ def run(self):
 
     seconds = staticmethod(runtimeSeconds)
 
-    def callLater(self, _seconds, _f, *args, **kw):
+    # We need to ignore mypy errors here because the '_f' parameter is
+    # not in IReactorTime.callLater
+    def callLater(self, _seconds, _f, *args, **kw):  # type: ignore[override]
```
Comment (adiroiban) [inline src/twisted/internet/base.py]: just use the signature from the interface.

File: src/twisted/pair/ethernet.py [hunk 3]
```diff
@@ -15,14 +15,18 @@
 from zope.interface import implementer, Interface
 
 
+
 class IEthernetProtocol(Interface):
     """An interface for protocols that handle Ethernet frames"""
-    def addProto():
+    def addProto(num, proto):
```
Comment (adiroiban) [inline src/twisted/pair/ethernet.py:21]: this interface change looks good can can be done here... but maybe use a `.remove` release note to let other know that the IEthernetProtocol interface was updated to match all the existing implementations.

**Review by adiroiban — CHANGES_REQUESTED** [review]
It looks good.

For addSystemEventTrigger and callLater I think that the implementation should just use the same argument names as defined in the interface.
I have no idea why *private* arguments were used.

Thanks again.

File: src/twisted/internet/base.py (hunk 2 above)
Comment (rodrigc) [reply src/twisted/internet/base.py]: How?  The signature from the interface takes 4 args, but this function signature takes 5 args.

Comment (adiroiban) [inline src/twisted/internet/base.py]: Which is the 5th argument?

This change would look valid to me:

```suggestion
    def callLater(self, delay, callable, *args, **kw):]
```

This is based on

```
src/twisted/internet/interfaces.py:    def callLater(delay, callable, *args, **kw):
```

File: src/twisted/internet/base.py (hunk 1 above)
Comment (rodrigc) [reply src/twisted/internet/base.py]: I've moved this to: https://github.com/twisted/twisted/pull/1325

File: src/twisted/internet/base.py (hunk 2 above)
Comment (rodrigc) [reply src/twisted/internet/base.py]: I've moved this to https://github.com/twisted/twisted/pull/1325

Comment (rodrigc) [reply src/twisted/internet/base.py]: I've moved this to https://github.com/twisted/twisted/pull/1325

File: src/twisted/pair/ethernet.py (hunk 3 above)
Comment (rodrigc) [reply src/twisted/pair/ethernet.py:21]: I added newsfragment

**Review by adiroiban — APPROVED** [review]
Looks good. Thanks!

I think that  src/twisted/newsfragments/9877.misc file can be removed.

**Pull request merged**